### PR TITLE
An exception handler has been added

### PR DESCRIPTION
**Pull request by Mark0VLK**

### CODE REVIEW

File: src/main/java/com/ieltswise/exception/EventFetchingException.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.ieltswise.exception;
+
+import java.io.IOException;
+
+public class EventFetchingException extends IOException {
+
+    public EventFetchingException(String msg) {
```
Comment (tihonk) [inline src/main/java/com/ieltswise/exception/EventFetchingException.java]: Next time, if you call a variable “message” in an exception class, then use this name everywhere.

File: src/main/java/com/ieltswise/exception/handler/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        logError(e);
+        return new ResponseEntity<>(
+                new ErrorMessage(
+                        PAYPAL_ERROR.getCodeId(),
+                        e.getMessage()
+                ),
+                HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @ExceptionHandler(IllegalArgumentException.class)
```
Comment (tihonk) [inline src/main/java/com/ieltswise/exception/handler/GlobalExceptionHandler.java:89]: Great, I like this approach.

File: src/main/java/com/ieltswise/service/BookingService.java [hunk 3]
```diff
@@ -1,35 +1,42 @@
...
     /**
      * A method for booking a regular lesson with a tutor in Google Calendar
-     * @param sessionData contains information about the participants and the time of the planned event
+     *
+     * @param regularSessionDataRequest an object containing data for requesting a regular lesson
      * @return booking session data
+     * @throws Exception if an error occurs during the booking process
      */
-    SessionDataResponse bookRegularSession(final SessionDataRequest sessionData);
+    SessionDataResponse bookRegularSession(final RegularSessionDataRequest regularSessionDataRequest) throws Exception;
```
Comment (tihonk) [inline src/main/java/com/ieltswise/service/BookingService.java:26]: Somnitelьno no okэй

File: src/main/java/com/ieltswise/service/impl/GoogleEventsServiceImpl.java [hunk 4]
```diff
@@ -58,24 +61,31 @@ public class GoogleEventsServiceImpl implements GoogleEventsService {
...
+        isTutorRegistered(tutorID);
         try {
             URL obj = new URL("https://www.googleapis.com/calendar/v3/calendars/" + tutorID
                     + "/events?key=" + googleCredentialKey);
             return extractEvents(createJSONObjectResponse(obj).getJSONArray(ITEMS));
-        } catch (Exception e) {
+        } catch (IOException e) {
             log.error("Failed to fetch events for tutor ID: {}", tutorID, e);
-            throw new RuntimeException(e);
+            throw new EventFetchingException(e.getMessage());
```
Comment (tihonk) [inline src/main/java/com/ieltswise/service/impl/GoogleEventsServiceImpl.java:81]: Great fix

File: src/test/java/com/ieltswise/controller/CommentControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+                .id(1L)
+                .name("Bob")
+                .email("Bob001@gmail.com")
+                .value("I like learning English.")
+                .created(LocalDateTime.of(2024, 5, 2, 20, 19, 35, 0))
+                .build();
+    }
+
+    @Test
+    public void CommentController_GetAllComments_ReturnAllComments() throws Exception {
```
Comment (tihonk) [inline src/test/java/com/ieltswise/controller/CommentControllerTest.java]: You need to follow the method naming style. In our case, this is a camelcase and everything unnecessary needs to be removed:
testGetAllCommentsReturnAllComments

**Review by tihonk — COMMENTED** [review]
In fact, unit tests are rarely written for controllers. For these, it is preferable to write integration ones. So overall everything is good. Now I'll check that everything is working.

File: src/main/java/com/ieltswise/exception/EventFetchingException.java (hunk 1 above)
Comment (Mark0VLK) [reply src/main/java/com/ieltswise/exception/EventFetchingException.java]: fixed it

File: src/main/java/com/ieltswise/exception/handler/GlobalExceptionHandler.java (hunk 2 above)
Comment (Mark0VLK) [reply src/main/java/com/ieltswise/exception/handler/GlobalExceptionHandler.java:89]: thanks

File: src/main/java/com/ieltswise/service/impl/GoogleEventsServiceImpl.java (hunk 4 above)
Comment (Mark0VLK) [reply src/main/java/com/ieltswise/service/impl/GoogleEventsServiceImpl.java:81]: thanks

File: src/test/java/com/ieltswise/controller/CommentControllerTest.java (hunk 5 above)
Comment (Mark0VLK) [reply src/test/java/com/ieltswise/controller/CommentControllerTest.java]: fixed it

**Review by tihonk — APPROVED** [review]

**Pull request merged**